bin/compile

```diff
@@ -157,8 +157,13 @@ install_bins() {
   echo ""
 
   warn_node_engine "$node_engine"
+
+  meta_set "build-step" "install-nodejs"
   monitor "install-node-binary" install_nodejs "$node_engine" "$BUILD_DIR/.heroku/node"
+
+  meta_set "build-step" "install-npm"
```

Comment:
Apologies for the noob question - I can see `meta_set` is used frequently for executing steps, but what is it exactly doing?

Reply:
`meta_set` writes to a key-value store, so we're over-writing a key called "build-step" each time

https://github.com/heroku/heroku-buildpack-nodejs/blob/master/lib/metadata.sh
https://github.com/heroku/heroku-buildpack-nodejs/blob/master/lib/kvstore.sh

The values get logged out at the end of the build, and also whenever there's a failure:

https://github.com/heroku/heroku-buildpack-nodejs/blob/master/bin/compile#L322

This allows us to distinguish between builds that failed installing Node (likely our fault, high significance) and builds that failed running the user's build step (failures are common and we'll never eliminate them, lower significance).

As an example, you can get a breakdown of failures are distributed across the various build steps in splunk by searching: `buildpack=nodejs node_build_success=false | top limit=20 build_step`

Reply:
Thanks for the explanation 👍